cluster/v1/types.go

```diff
@@ -66,6 +66,18 @@ type ManagedClusterSpec struct {
 	// If its value is zero, the Klusterlet agent will update its lease every 60s by default
 	// +optional
 	LeaseDurationSeconds int32 `json:"leaseDurationSeconds,omitempty"`
+
+	// ClusterSet is the name of the ManagedClusterSet to which the ManagedCluster belongs to.
+	// If the value is empty, the ManagedCluster does not belong to any ManagedClusterSet.
+	// The following conditions must be met in order to set the value to a certain ManagedClusterSet:
+	//   1. This ManagedClusterSet must exist;
+	//   2. User must have a RBAC rule to UPDATE on this ManagedClusterSet;
+	//   3. If it already refers to a ManagedClusterSet, user must have a RBAC rule to UPDATE on that
+	//      ManagedClusterSet either.
+	// If the value is set, deletion of the ManagedClusterSet will also trigger deletion of the containing
```

Comment:
We normally allow dangling references so that we have eventual consistency and don't have to handle referential integrity problems that result in user resource deletion.

Reply:
This is the one property I am still a bit iffy about. I guess we at least are starting with the option that is easier to relax. It seems like it would be easier to begin allowing dangling references than to begin forbidding them. Let's pay careful attention to any feedback we get from users about this property.